extensions/sql-database-projects/src/dialogs/updateProjectFromDatabaseDialog.ts

```diff
@@ -0,0 +1,610 @@
+
+	private async getServerValues() {
+		let cons = await azdata.connection.getConnections(/* activeConnectionsOnly */ true);
+
+		// This user has no active connections
+		if (!cons || cons.length === 0) {
+			return undefined;
+		}
+
+		// Update connection icon to "connected" state
+		this.connectionButton!.iconPath = IconPathHelper.connect;
+
+		// reverse list so that most recent connections are first
+		cons.reverse();
+
+		let count = -1;
+		let idx = -1;
+		let values = cons.map(c => {
+			count++;
+
+			let usr = c.options.user;
+
+			if (!usr) {
+				usr = constants.defaultUser;
+			}
+
+			let srv = c.options.server;
+
+			let finalName = `${srv} (${usr})`;
+
+			if (c.options.connectionName) {
+				finalName = c.options.connectionName;
+			}
+
+			if (c.connectionId === this.connectionId) {
+				idx = count;
+			}
+
+			return {
+				connection: c,
+				displayName: finalName,
+				name: srv,
+			};
+		});
+
+		// move server of current connection to the top of the list so it is the default
+		if (idx >= 1) {
+			let tmp = values[0];
+			values[0] = values[idx];
+			values[idx] = tmp;
+		}
+
+		values = values.reduce((uniqueValues: { connection: azdata.connection.ConnectionProfile, displayName: string, name: string }[], conn) => {
+			let exists = uniqueValues.find(x => x.displayName === conn.displayName);
+			if (!exists) {
+				uniqueValues.push(conn);
+			}
+			return uniqueValues;
+		}, []);
+
+		return values;
+	}
+
+	protected async getDatabaseValues(connectionId: string) {
+		let idx = -1;
+		let count = -1;
+
+		let values = (await azdata.connection.listDatabases(connectionId)).sort((a, b) => a.localeCompare(b)).map(db => {
+			count++;
+
+			// put currently selected db at the top of the dropdown if there is one
+			if (this.profile && this.profile.databaseName && this.profile.databaseName === db) {
+				idx = count;
+			}
+
+			return db;
+		});
+
+		if (idx >= 0) {
+			let tmp = values[0];
+			values[0] = values[idx];
+			values[idx] = tmp;
+		}
+		return values;
+	}
+
+	private createConnectionButton(view: azdata.ModelView) {
+		this.connectionButton = view.modelBuilder.button().withProps({
+			ariaLabel: constants.selectConnection,
+			iconPath: IconPathHelper.selectConnection,
+			height: '20px',
+			width: '20px'
+		}).component();
+
+		this.connectionButton.onDidClick(async () => {
+			await this.connectionButtonClick();
+			this.connectionButton!.iconPath = IconPathHelper.connect;
+		});
+	}
+
+	private async connectionButtonClick() {
+		let connection = await azdata.connection.openConnectionDialog();
+		if (connection) {
+			this.connectionId = connection.connectionId;
+			this.populateServerDropdown();
+		}
+	}
+
+	private createProjectLocationRow(view: azdata.ModelView): azdata.FlexContainer {
+		const browseFolderButton: azdata.Component = this.createBrowseFileButton(view);
+
+		const value = this.project ? this.project.projectFilePath : '';
+
+		this.projectFileTextBox = view.modelBuilder.inputBox().withProps({
```

Comment:
has having a dropdown with the open projects(like the add database reference dialog) instead of an input box with browse button been considered?

Reply:
Don't know that it has.  I wouldn't mind swapping to a dropdown with "Browse..." at the end because I think there's some value to being able to target an arbitrary csproj, but maybe that change to a dropdown can happen in a different check-in?